light/example_test.go

```diff
@@ -81,91 +79,33 @@ func ExampleClient_Update() {
 		}
 	}()
 
-	time.Sleep(2 * time.Second)
-
-	h, err := c.Update(ctx, time.Now())
-	if err != nil {
-		stdlog.Fatal(err)
-	}
-
-	if h != nil && h.Height > 2 {
-		fmt.Println("successful update")
-	} else {
-		fmt.Println("update failed")
-	}
-}
-
-// Manually getting light blocks and verifying them.
-func ExampleClient_VerifyLightBlockAtHeight() {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	conf := rpctest.CreateConfig("ExampleClient_VerifyLightBlockAtHeight")
-
-	// Start a test application
-	app := kvstore.NewApplication()
-
-	_, closer, err := rpctest.StartTendermint(ctx, conf, app, rpctest.SuppressStdout)
+	// wait for a few more blocks to be produced
+	_, err = waitForBlock(ctx, primary, 4)
 	if err != nil {
 		stdlog.Fatal(err)
 	}
-	defer func() { _ = closer(ctx) }()
-
-	// give Tendermint time to generate some blocks
-	time.Sleep(5 * time.Second)
-
-	dbDir, err := ioutil.TempDir("", "light-client-example")
-	if err != nil {
-		stdlog.Fatal(err)
-	}
-	defer os.RemoveAll(dbDir)
-
-	chainID := conf.ChainID()
 
-	primary, err := httpp.New(chainID, conf.RPC.ListenAddress)
-	if err != nil {
-		stdlog.Fatal(err)
-	}
-
-	block, err := primary.LightBlock(ctx, 2)
-	if err != nil {
-		stdlog.Fatal(err)
-	}
-
-	db, err := dbm.NewGoLevelDB("light-client-db", dbDir)
+	// veify the block at height 3
+	_, err = c.VerifyLightBlockAtHeight(context.Background(), 3, time.Now())
 	if err != nil {
 		stdlog.Fatal(err)
 	}
 
-	c, err := light.NewClient(ctx,
-		chainID,
-		light.TrustOptions{
-			Period: 504 * time.Hour, // 21 days
-			Height: 2,
-			Hash:   block.Hash(),
-		},
-		primary,
-		[]provider.Provider{primary}, // NOTE: primary should not be used here
-		dbs.New(db),
-		light.Logger(log.TestingLogger()),
-	)
+	// retrieve light block at height 3
+	h, err := c.TrustedLightBlock(3)
 	if err != nil {
 		stdlog.Fatal(err)
 	}
-	defer func() {
-		if err := c.Cleanup(); err != nil {
-			stdlog.Fatal(err)
-		}
-	}()
 
-	_, err = c.VerifyLightBlockAtHeight(context.Background(), 3, time.Now())
+	// update to the latest height
+	h, err = c.Update(ctx, time.Now())
 	if err != nil {
 		stdlog.Fatal(err)
 	}
 
-	h, err := c.TrustedLightBlock(3)
-	if err != nil {
-		stdlog.Fatal(err)
+	if h != nil && h.Height > 3 {
+		fmt.Println("successful update")
```

Comment:
Is there a reason these a printlns and not log statements?

Reply:
wasn't my decision. I can switch to using the logger

Reply:
Let's use a logger if we can.